activesupport/test/testing/callbacks_test.rb

```diff
@@ -34,3 +34,23 @@ def test_teardown_raise_but_all_after_teardown_method_are_called
     assert true
   end
 end
+
+class AroundCallbackTest < ActiveSupport::TestCase
+  class Client
+    class_attribute :stubbed, default: false
+  end
+
+  around do |test, block|
+    assert_not Client.stubbed
+
+    Client.stubbed = true
+    block.call
+    Client.stubbed = false
```

Comment:
As you mentioned in https://github.com/rails/rails/pull/50889#discussion_r1468029174, it's a good idea to add a failing test and ensure this runs.

You can create a inner test and assert in the outer test that it failed

```ruby
  test "around block when a test fails" do
    failing_test = Class.new(ActiveSupport::TestCase) do
      attr_reader :witness

      around do |_, test|
        test.call

        @witness = true
      end

      def self.name
        "FailingTest"
      end

      def test_is_failing
        assert(false)
      end
    end.new("test_is_failing")

    assert_nil(failing_test.witness)
    result = failing_test.run

    refute_predicate(result, :passed?)
    assert(failing_test.witness)
  end
```

Reply:
I've pushed up a change to incorporate this test. I've also added you as a co-author. Thank you!